mslib/msui/mscolab.py

```diff
@@ -219,6 +219,10 @@ def connect_handler(self):
             s.auth = auth
             s.headers.update({'x-test': 'true'})
             r = s.get(url_join(url, 'status'), timeout=(2, 10))
+            try:
+                idp_enabled = json.loads(r.text)["IDP_ENABLED"]
+            except (json.decoder.JSONDecodeError, KeyError):
+                idp_enabled = False
```

Comment:
Since this is handling the "happy-path", i.e. a successful request to the `/status` endpoint, this should be further down in the if-branch checking `r.status_code == 200`. A good spot should be right before the `if idp_enabled:` line.